aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java

```diff
@@ -296,6 +323,50 @@ public void testSignUp() throws Exception {
         assertEquals("Cannot support MFA in tests", SignInState.DONE, signInResult.getSignInState());
     }
 
+    /**
+     * When the user signs in with temporary credentials, they should be prompted to set a new password.
+     * To do this, they can supply a map of attributes to {@link confirmSignIn(....)},
+     * including the challenge response (key = NEW_PASSWORD), as well as
+     * user attributes (key = userAttributes.<attribute name>) to be
+     * set as part of the update. The ability to pass additional attributes is
+     * needed so the caller can pass in any required attributes
+     * other than email and/or mobile.
+     */
+    @Test
+    public void testSignInForceChangePassword() throws Exception{
+        final CountDownLatch stateNotificationLatch = new CountDownLatch(1);
+        final AtomicReference<UserStateDetails> userState = new AtomicReference<UserStateDetails>();
+        listener = new UserStateListener() {
+            @Override
+            public void onUserStateChanged(UserStateDetails details) {
+                userState.set(details);
+                auth.removeUserStateListener(listener);
+                stateNotificationLatch.countDown();
+            }
+        };
+        auth.addUserStateListener(listener);
+        //Sign-in using the temporary password
+        final SignInResult signInResult = auth.signIn(USERNAME_ADMIN_API_USER, TEMP_PASSWORD, null);
+        //Assert we're receiving challenge to change password
+        assertEquals(NEW_PASSWORD_REQUIRED, signInResult.getSignInState());
+
+        HashMap<String, String> inputAttributes = new HashMap<String, String>();
+        //Add attribute to set new password
+        inputAttributes.put(CHALLENGE_RESPONSE_NEW_PASSWORD_KEY, PASSWORD);
+        //Set some extra attributes
+        inputAttributes.put(CHALLENGE_RESPONSE_USER_ATTRIBUTES_PREFIX_KEY + "name", "dummy-name");
```

Comment:
Having the separate constant for _prefix_ is still a little weird and Cognito-y.

Can we just have two simple constants like `CHALLENGE_RESPONSE_USER_NAME_KEY`? Or even better, if the string value doesn’t have anything specific to the challenge response, maybe just `USER_NAME_KEY`?